### DISCUSSION TITLE
Question: Caching

### DISCUSSION THREAD

**Discussion by ErSpieltHD** [AdguardTeam/AdGuardHome#4002] [Q&A] [answered]
Hey there,
is optimistic caching better/faster than leaving it disabled? How much storage does use and how many DNS entries are cached? When are they overwritten?

I don't know where else to ask this so yeah.

**Comment by EugeneOne1** [comment, marked as answer]
Please use **Discussions** section for such kind of questions.

The primary purpose of the optimistic cache is to respond with expired cache entries while refreshing those in the background.  Thus it should reduce the time spent on resolving responses on demand.  However, it introduces the possibility of outdated response data if the actual DNS record at the upstream server has been updated.  That is why such responses having a diminished TTL of 60 seconds.  In general, optimistic caching feature doesn't increase the resources used since it works with the same storage the common cache works with.

The number of cached DNS entries is controlled by the **Cache size** setting in **DNS settings** section.  It's hard to tell how much entries are stored exactly since the size of cache is specified in bytes and cached DNS messages may vary by their size.  However, the maximum length of a message is 65536 bytes, so the maximum number of such messages stored in cache of default size of 4194304 bytes is about 64.  In real world this number is much larger though.

**Reply by ErSpieltHD** [reply]
Hey,

thx for all the info. I've read it as soon as you wrote it but just didn't reply.

How can I clear the DNS cache? On Windows you just do ipconfig /flushdns

**Reply by EugeneOne1** [reply]
AdGuard Home currently doesn't explicitly allow to clear the cache, this is already a [feature request](https://github.com/AdguardTeam/AdGuardHome/issues/3146).

However, the cache is cleared after the following actions:

 *  changing the filters (adding/removing rules and lists);
 *  changing the upstream configuration;
 *  restarting the AGH.
